src/main/java/org/jenkinsci/plugins/workflow/cps/CpsGroovyShell.java

```diff
@@ -127,7 +127,15 @@ public Script parse(GroovyCodeSource codeSource) throws CompilationFailedExcepti
     }
 
     private Script doParse(GroovyCodeSource codeSource) throws CompilationFailedException {
-        try (GroovySandbox.Scope scope = new GroovySandbox().enter()) {
+        GroovySandbox sandbox = new GroovySandbox();
+        if (execution != null) {
+            sandbox.withWhitelist(new GroovyClassLoaderWhitelist(CpsWhitelist.get(),
```

Comment:
See https://github.com/jenkinsci/workflow-cps-plugin/pull/297#issuecomment-502818187 for discussion of the switch from the implicit `Whitelist.all()` to `CpsWhitelist.get()`. TL;DR: I think it doesn't really matter, the things that would be affected are broken either way.

Reply:
Ok, so parsing happens after the CPS transformation, but doesn't run in the CPS VM thread, so calling any CPS-transformed method during script parsing as in this example will fail: 

```
    @Issue("JENKINS-56682")
    @Test
    public void scriptInitializersClassSyntaxWithMethodCall() throws Exception {
        WorkflowJob p = jenkins.jenkins.createProject(WorkflowJob.class, "p");
        p.setDefinition(new CpsFlowDefinition(
                "class MyScript extends org.jenkinsci.plugins.workflow.cps.CpsScript {\n" +
                "  static { bar() }\n" +
                "  static int foo = 0\n" +
                "  static def bar() { MyScript.foo = MyScript.foo + 1 }\n" +
                "  def run() {\n" +
                "    echo(/MyScript.foo is ${MyScript.foo}/)\n " +
                "  }\n" +
                "}\n", true));
        WorkflowRun b = jenkins.buildAndAssertSuccess(p);
        jenkins.assertLogContains("MyScript.foo is 1", b);
    }
```
No matter the choice of whitelist, this fails with:

> CpsCallableInvocation{methodName=bar, call=com.cloudbees.groovy.cps.impl.CpsFunction@20907d04, receiver=class MyScript, arguments=[]}

I am going to switch back to `Whitelist.all()` since I don't think there is any case where `CpsWhitelist.get()` makes sense during script parsing.

Reply:
Done in b3aebc7.

Reply:
Anyway based on https://github.com/jenkinsci/workflow-cps-plugin/blob/7eab5dc0c39289a372a05904d65596c4192c4005/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsWhitelist.java#L44-L48 I suspect that use of `CpsWhitelist` here would have introduced vulnerabilities.